packages/approval-controller/src/ApprovalController.ts

```diff
@@ -91,7 +92,7 @@ export type ApprovalRequest<RequestData extends ApprovalRequestData> = {
   /**
    * The type of the approval request.
    */
-  type: string;
+  type: ApprovalType;
```

Comment:
Unfortunately, not every approval type matches our `ApprovalType` enum as this was added after in an effort to centralise the definitions across the clients.

There are many types in the extension that are not defined in this enum so unfortunately it wouldn't be accurate to assume this was always an `ApprovalType`.

We could push for every type to be defined here first then make this change, but it forces a anwkward workflow as new confirmations need to be added to the enum before they are implemented.

Reply:
My main concern now is we're introducing an additional large dependency to the package, but gaining very little as the `string` type already encompasses all the `ApprovalType` values.

The `ToDo` is a good reminder for this, but I don't think the `controller-utils` dependency is warranted at this point.

Plus if we fully commit to `ApprovalType` in future, we'd also likely want to update the validations to ensure new approval requests provide valid types etc.

Reply:
Great points and I agree. I've reverted the changes and left the TODO comment in. Thanks for the inputs @matthewwalsh0, @MajorLift, and @mcmire!